examples/srl/main_train_srl.py

```diff
@@ -0,0 +1,207 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Iterator, Dict, List
+from texar.torch.data import Batch
+from torch import Tensor
+from torch.optim import SGD
+from torch.optim.optimizer import Optimizer
+
+from forte.data.converter.feature import Feature
+from forte.data.extractor.attribute_extractor import AttributeExtractor
+from forte.data.extractor.base_extractor import BaseExtractor
+from forte.data.extractor.char_extractor import CharExtractor
+from forte.data.extractor.link_extractor import LinkExtractor
+from forte.train_preprocessor import TrainPreprocessor
+from forte.data.readers.ontonotes_reader import OntonotesReader
+from forte.models.srl_new.model import LabeledSpanGraphNetwork
+from forte.models.srl_new import data
+from ft.onto.base_ontology import Sentence, Token, PredicateLink
+
+logger = logging.getLogger(__name__)
+
+logging.basicConfig(level=logging.INFO)
+
+
+def create_model(schemes: Dict[str, Dict[str, BaseExtractor]]) -> \
+        LabeledSpanGraphNetwork:
+    text_extractor: BaseExtractor = schemes["text_tag"]["extractor"]
+    char_extractor: BaseExtractor = schemes["char_tag"]["extractor"]
+    link_extractor: BaseExtractor = schemes["pred_link_tag"]["extractor"]
+
+    _model = LabeledSpanGraphNetwork(word_vocab=text_extractor.get_dict(),
+                                     char_vocab_size=char_extractor.size(),
+                                     label_vocab=link_extractor.get_dict())
+
+    return _model
+
+
+def train(_model: LabeledSpanGraphNetwork,
+          _optim: Optimizer,
+          _batch: Batch) -> \
+        LabeledSpanGraphNetwork.ReturnType:
+    char_tensor: Tensor = _batch["char_tag"]["data"]
+    char_masks: List[Tensor] = _batch["char_tag"]["masks"]
+    text_tensor: Tensor = _batch["text_tag"]["data"]
+    text_mask: Tensor = _batch["text_tag"]["masks"][0]
+    text: List[List[str]] = _batch["raw_text_tag"]["data"]
+    pred_link_features: List[Feature] = _batch["pred_link_tag"]["features"]
+
+    _optim.zero_grad()
+
+    output: LabeledSpanGraphNetwork.ReturnType = \
+        model(text=text,
+              char_batch=char_tensor,
+              char_masks=char_masks,
+              text_batch=text_tensor,
+              text_mask=text_mask,
+              srl_features=pred_link_features)
+
+    output["loss"].backward()
+    optim.step()
+
+    return output
+
+
+def predict_forward_fn(_model: LabeledSpanGraphNetwork, _batch: Dict) -> Dict:
+    char_tensor: Tensor = _batch["char_tag"]["data"]
+    char_masks: List[Tensor] = _batch["char_tag"]["masks"]
+    text_tensor: Tensor = _batch["text_tag"]["data"]
+    text_mask: Tensor = _batch["text_tag"]["masks"][0]
+    text: List[List[str]] = _batch["raw_text_tag"]["data"]
+
+    # TODO: test enable enforce_constriant
```

Comment:
Are these todos going to be addressed this time? If not, we need better documentation here to make this TODO readable.